.github/workflows/deploy_docs.yml

```diff
@@ -0,0 +1,78 @@
+name: Deploy Docs on Release
+
+on:
+  push:
+    branches:
+      - main
+  release:
+    types: [ published ]
+    branches:
+      - main
+
+permissions:
+  contents: write
+
+jobs:
+  deploy-docs:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v3
```

Comment:
could we use v4 here?

Reply:
yes! i missed there was a newer version available